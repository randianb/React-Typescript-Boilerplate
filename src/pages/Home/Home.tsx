import React, { useRef } from 'react';
import { Redirect } from 'react-router';
import { IFormInputValues, getFormValues } from '../Login/Auth'
import Alert from '../../components/Alert'
import Table from '../../components/Table'

export default function HomePage() {
  const [values] = React.useState<IFormInputValues>(getFormValues(false));
  const childRef = useRef<{openModal:()=>{}}>(null!);

  if (values.token.length > 0) {
    return (
      <main className="home">
        <h1>Welcome</h1>

        <Alert title='抽检结果' message='抽检正在进行中' confirmtxt='确认' ref={childRef}></Alert>
        <button onClick={() => childRef.current.openModal()}>Click Parent</button>
        <Table></Table>
      </main>
    );
  }
  else {
    return <Redirect to="/login"></Redirect>
  }
}
export class Home extends React.Component {

  public render() {
    return <HomePage></HomePage>
  }
}


