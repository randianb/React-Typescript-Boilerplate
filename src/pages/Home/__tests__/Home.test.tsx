import React from 'react';
import Home from '../Home';
import { render, cleanup } from 'react-testing-library';

afterEach(cleanup);

describe('Home', () => {
  it('Renders Home page', () => {
    const { container } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
