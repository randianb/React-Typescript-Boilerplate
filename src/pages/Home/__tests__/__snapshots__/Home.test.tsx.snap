// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renders Home page 1`] = `
<main
  class="home"
>
  <h1>
    Welcome
  </h1>
</main>
`;
